core/src/main/java/org/hibernate/ogm/datastore/impl/AvailableDatastoreProvider.java

```diff
@@ -19,6 +19,7 @@
 	EHCACHE( "org.hibernate.ogm.datastore.ehcache.impl.EhcacheDatastoreProvider" ),
 	MONGODB( "org.hibernate.ogm.datastore.mongodb.impl.MongoDBDatastoreProvider" ),
 	FONGO( "org.hibernate.ogm.datastore.mongodb.impl.FongoDBDatastoreProvider" ),
+	NEO4J( "org.hibernate.ogm.datastore.neo4j.embedded.impl.EmbeddedNeo4jDatastoreProvider" ),
```

Comment:
I sort of like the idea of having a generic variable and choose the dialect based on what we have on the classpath and or properties. Sort of like if you set the database location to /opt/database or http:// localhost:7777 or bolt://...

Unless, we plan of remove it I guess we can wait a bit before deprecating it